### ISSUE TITLE
ERR_INVALID_ARG_TYPE

### ISSUE THREAD

**Issue by fernandomarca** [bubao/Max6675-Raspberry-pi-nodejs#4]
Hi, hello i am getting this error when running the example code. The Error says that parameter CS=4 must be of type string.
Any idea what could be happening? Thanks

```js
const CS = 4;
const SCK = 24;
const SO = 25;
const UNIT = 1;
const max6675 = new Max6675();
max6675.setPin(CS, SCK, SO, UNIT);
```
```bash
node:internal/errors:465
ErrorCaptureStackTrace(err);
    ^
**TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received type number (4)**
    at Object.writeFileSync (node:fs:2163:5)
    at new Gpio (/home/fernando/Documentos/GitHub/FullCycle/Curso/01-docker/node-red/node_modules/onoff/onoff.js:96:10)
    at Max6675.setPin (/home/fernando/Documentos/GitHub/FullCycle/Curso/01-docker/node-red/node_modules/max6675-raspi/index.js:87:14)
    at new Max6675 (/home/fernando/Documentos/GitHub/FullCycle/Curso/01-docker/node-red/node_modules/max6675-raspi/index.js:22:9)
    at file:///home/fernando/Documentos/GitHub/FullCycle/Curso/01-docker/node-red/index.js:7:17
    at ModuleJob.run (node:internal/modules/esm/module_job:198:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:385:24)
    at async loadESM (node:internal/process/esm_loader:88:5)
    at async handleMainPromise (node:internal/modules/run_main:61:12) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

**Comment by bubao** [comment]
hi,I fixed the bug in [fix gpio fs.writeFileSync data must be string or buffer](https://github.com/bubao/Max6675-Raspberry-pi-nodejs/commit/6e629fcc0a2e9587bf729e97eb2e836650fb95b3) commit.But I don't have a Raspberry Pi for testing.So,I will publish a new version in npm when you test it successfully。

**Comment by marklise** [comment]
> hi,I fixed the bug in [fix gpio fs.writeFileSync data must be string or buffer](https://github.com/bubao/Max6675-Raspberry-pi-nodejs/commit/6e629fcc0a2e9587bf729e97eb2e836650fb95b3) commit.But I don't have a Raspberry Pi for testing.So,I will publish a new version in npm when you test it successfully。

You only need to change the "" strings to fix the bug.  The capitalization is irrelevant and should not be modified.

```
this.cs = new Gpio(this.cs + "", "low");
this.sck = new Gpio(this.sck + "", "low");
this.so = this.so.map(item => new Gpio(item + "", "in"));
```

I have verified this fixes the issue above with a PI model 4B.

However, there is another issue once this fix is in place:

```
[20.07.2022 12:13.52.910] [ERROR] Error: EBUSY: resource busy or locked, write
    at Object.writeSync (node:fs:884:3)
    at Object.writeFileSync (node:fs:2159:26)
    at new Gpio (/home/pi/src/test/node_modules/max6675-raspi/node_modules/onoff/onoff.js:96:10)
    at Max6675.setPin (/home/pi/src/test/node_modules/max6675-raspi/index.js:87:14)
    at Object.<anonymous> (/home/pi/src/test/modules/MMM-Test/node_helper.js:9:9)
    at Module._compile (node:internal/modules/cjs/loader:1116:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1169:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:829:12)
    at Function.c._load (node:electron/js2c/asar_bundle:5:13343)
```

**Comment by marklise** [comment]
Actually - this is due to the one-wire interface, as it uses GPIO 4 (pin 7) as the default interface.  I changed it to use GPIO 17 and it no longer throws any errors.
